Logs API to be user-facing (#4225)
open-telemetry/opentelemetry-specification: specification/logs/event-api.md

# Events API

**Status**: [Development](../document-status.md)

<details>
<summary>Table of Contents</summary>

<!-- Re-generate TOC with `markdown-toc --no-first-h1 -i` -->

> <!-- toc -->

Reviewer: This file must either be merged into the logs or kept separate

Author: Kept separate.

  provides access to `EventLogger`s.
* [EventLogger](#eventlogger) is the component responsible for emitting events.

## Event API use cases

> The Events API was designed to allow shared libraries to emit
> [events](data-model.md#events) without needing to depend on a third party logger.

Reviewer 2: Perhaps this part is changing now. 

Maybe

```suggestion
The Events API was designed to allow shared libraries and applications to emit named and structured logs with consistent semantics.
```

?

Or maybe just remove this sentence?

Author: LGTM. We can the apply your suggestion if there are no counter arguments.

Reviewer: It was designed to emit named and structured EVENTS that happen to be sent / represented as a `LogRecord`, they are still EVENTS and not JUST A LOG!

Reviewer 2: @MSNev can you make a suggestion that would work for you? The original sentence is no longer applicable, what would  be a good replacement?

Author: But they are still logs 😉

Reviewer: Why is it no longer applicable, the original sentence is still valid. If the Events API is being merged into a new common Log API then nothing has changed.

Author: I agree with @MSNev that the current sentence is still applicable/valid. @lmolkova, how about resolving?

Reviewer 2: I think it's not relevant since with this PR we will have user-facing logging API.

I.e. we have user-facing logging API and the second part of this sentence no longer makes sense: "The Events API was designed to allow shared libraries to emit events __without needing to depend on a third party logger__"

Author: Maybe

```suggestion
The Events API was designed to allow shared libraries to emit
[events](data-model.md#events).
```

> Instrumentation authors and application developers are encouraged can call this

Reviewer 3: ```suggestion
Instrumentation authors and application developers are encouraged to call this
```
